controllers/workspace/workspace_controller.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Red Hat
+
+package workspace
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+
+	giterrors "github.com/pkg/errors"
+	"github.com/stolostron/cluster-registration-operator/deploy"
+	"github.com/stolostron/cluster-registration-operator/pkg/helpers"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	clusteradmapply "open-cluster-management.io/clusteradm/pkg/helpers/apply"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// WorkspaceReconciler reconciles namespaces with workspace annotation
+type WorkspaceReconciler struct {
+	client.Client
+	KubeClient         kubernetes.Interface
+	DynamicClient      dynamic.Interface
+	APIExtensionClient apiextensionsclient.Interface
+	Log                logr.Logger
+	Scheme             *runtime.Scheme
+	MceClusters        []helpers.MceInstance
+	// MceKubeClient         kubernetes.Interface
+	// MceDynamicClient      dynamic.Interface
+	// MceAPIExtensionClient apiextensionsclient.Interface
+}
+
+func (r *WorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Log.WithValues("namespace", req.Namespace, "name", req.Name)
+	logger.Info("Reconciling...")
+
+	//TODO - handle delete
+
+	if err := r.syncManagedClusterSet(req.Name, ctx); err != nil {
+		logger.Error(err, "failed to sync ManagedClusterSet")
+		//TODO - should we report a status on the namespace?
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *WorkspaceReconciler) syncManagedClusterSet(name string, ctx context.Context) error {
+	applierBuilder := &clusteradmapply.ApplierBuilder{}
+	applier := applierBuilder.WithClient(r.MceClusters[0].KubeClient, r.MceClusters[0].APIExtensionClient, r.MceClusters[0].DynamicClient).Build() //TODO - support more than one
+	readerDeploy := deploy.GetScenarioResourcesReader()
+
+	files := []string{
+		"workspace/managed_cluster_set.yaml",
+	}
+
+	values := struct {
+		Name string
+	}{
+		Name: name,
+	}
+
+	_, err := applier.ApplyCustomResources(readerDeploy, values, false, "", files...)
+	if err != nil {
+		return giterrors.WithStack(err)
+	}
+	return nil
```

Comment:
Do we want to check if managedclusterset is already exists before creating it ?

Reply:
The apply takes care of that.